translation/markdown.go

```diff
@@ -0,0 +1,217 @@
+				translationSeg = append(translationSeg, n.Segment)
+			}
+		}
+		// TODO: Skip for custom node in goplus markdown
+
+		return ast.WalkContinue, nil
+	})
+
+	// Translate
+	toBeTranslatedStr := strings.Join(translationVec, fmt.Sprintf("\n%s\n", translationSep))
+	translatedStr, err := c.TranslatePlainText(toBeTranslatedStr, from, to)
+	if err != nil {
+		return []byte(""), err
+	}
+
+	// Replace text
+	resultVec := strings.Split(translatedStr, translationSep)
+	if len(resultVec) != len(translationVec) {
+		return []byte(""), ErrTranslationNotMatch
+	}
+
+	result := make([]byte, 0)
+	result = append(result, src[:translationSeg[0].Start]...)
+	result = append(result, []byte(resultVec[0])...)
+	for idx := 1; idx < len(resultVec); idx++ {
+		result = append(result, src[translationSeg[idx-1].Stop:translationSeg[idx].Start]...)
+		result = append(result, []byte(strings.TrimSpace(resultVec[idx]))...)
+	}
+	result = append(result, src[translationSeg[len(translationSeg)-1].Stop:]...)
+
+	return result, nil
+}
+
+// TranslateMarkdown translate markdown text
+func (c *Engine) TranslateMarkdownText(src string, from, to language.Tag) (ret string, err error) {
+	retByte, err := c.TranslateMarkdown([]byte(src), from, to)
+
+	return string(retByte), err
+}
+
+func generateSeparator() string {
+	return fmt.Sprintf("%d", time.Now().Unix())
+}
+
+// Translate translate sequence of bytes
+func (c *Engine) TranslatePlain(src []byte, from, to language.Tag) (ret []byte, err error) {
+	retString, err := c.TranslatePlainText(string(src), from, to)
+
+	return []byte(retString), err
+}
+
+// Translate translate sequence of text
+func (c *Engine) TranslatePlainText(src string, from, to language.Tag) (ret string, err error) {
+	// Get translate result from api server
+	// Request form data
+	formData := url.Values{
+		"from":     {from.String()},
+		"to":       {to.String()},
+		"apikey":   {c.APIKey},
+		"src_text": {src},
+	}
+
+	var req *http.Request
+	req, err = http.NewRequest("POST", niuTransAPI, strings.NewReader(formData.Encode()))
+	if err != nil {
+		return "", ErrRequestFailed
+	}
+
+	// Set request headers
+	req.Header.Set("Content-Type", contentType)
+	req.Header.Set("User-Agent", userAgent)
+
+	if resp, err := http.DefaultClient.Do(req); err != nil {
+		return "", ErrRequestFailed
+	} else {
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			return "", ErrRequestFailed
+		}
+
+		// Parse response body
+		var raw []byte
+		if raw, err = io.ReadAll(resp.Body); err != nil {
+			return "", ErrRequestFailed
+		}
+
+		// Parse json result
+		var response TranslateResponse
+		if err = json.Unmarshal(raw, &response); err != nil {
+			return "", ErrRequestFailed
+		}
+
+		return response.TgtText, nil
+	}
+
+	return "", ErrRequestFailed
+}
+
+// TranslateBatchSeq translate a series of sequences of text
+// Deprecated: use TranslatePlainXXX instead
+func (c *Engine) TranslateBatchSeq(src []string, from, to language.Tag) ([]string, error) {
```

Comment:
remove this func

Reply:
Translation API has 5000 characters limit, so I need to split the text to batch plains.

I will update it to support long markdown text translation. @xushiwei

Reply:
batch can be a internal mechanism. `TranslatePlain` should have no characters limit.